src/components/checkbox/GroupableCheckboxReducers.ts

```diff
@@ -35,6 +35,16 @@ export const groupableCheckboxReducer = (state: IGroupableCheckboxesState = grou
                     ? toggleParentCheckbox(state, action)
                     : toggleChildCheckbox(state, action);
             }
+        case GroupableCheckboxActions.disabledGroup:
+            if (state.parentId === action.payload.id || state.parentId === action.payload.parentId) {
+                return action.payload.isParent
+                    ? disabledParentCheckbox(state, action)
+                    : disabledChildCheckbox(state, action);
+            }
+        case GroupableCheckboxActions.disabledAllGroup:
+            if (state.parentId === action.payload.id || state.parentId === action.payload.parentId) {
+                return disabledAllCheckbox(state, action);
+            }

```

Comment:
I really like your solution to simply end up in default instead of always have an else and return the same thing in each case.

Reply:
I agree this is very clean